Frameworks/MQTTnet.AspnetCore/ApplicationBuilderExtensions.cs

```diff
@@ -14,8 +15,18 @@ public static IApplicationBuilder UseMqttEndpoint(this IApplicationBuilder app,
             {
                 if (context.Request.Path == path && context.WebSockets.IsWebSocketRequest)
                 {
+                    string subprotocol = null;
+
+                    if (context.Request.Headers.TryGetValue("Sec-WebSocket-Protocol", out var requestedSubProtocolValues)
+                     && requestedSubProtocolValues.Count > 0
```

Comment:
Can you explain why this fixes the issue? I am no expert about websockets etc. so I can only guess 😄 Especially why this fails for UWP and netcore only!?

Reply:
it was just the part of the issue he has with establishing a connection. i just guessed this was the issue because i changed it when adding the test website. i verified with the testapp and voila problem solved. 

but that was just part of the issue the communicationtimeout issue persists.